executor/insert.go

```diff
@@ -81,8 +81,14 @@ func (e *InsertExec) exec(ctx context.Context, rows [][]types.Datum) error {
 			return err
 		}
 	} else {
-		for _, row := range rows {
-			if _, err := e.addRecord(ctx, row); err != nil {
+		for i, row := range rows {
+			var err error
+			if i == 0 {
+				_, err = e.addRecordWithAutoIDHint(ctx, row, len(rows))
```

Comment:
Do we not need to limit the size of reserveAutoIDCount?

Reply:
Unnecessary, the rows count equals to the size of a chunk https://github.com/pingcap/tidb/blob/master/executor/insert_common.go#L420

@Little-Wallace